lib/transaction.py

```diff
@@ -511,6 +511,32 @@ def is_schnorr_signed(self, input_idx):
                        for sig in self._inputs[input_idx].get('signatures', []))
         return False
 
+    def rpa_paycode_swap_dummy_for_destination(self,rpa_dummy_address, rpa_destination_address):
+        # This method was created for RPA - reusable payment address.  
+        # It swaps out a dummy destination address for the rpa-generated address.
+        # WARNING: This is not recommended for use outside of RPA since it could
+        # be dangerous to change the destination address of a transaction.
+ 
+        # This method expects cashaddr strings for parameters rpa_dummy_address and rpa_destination_address
+  
+        x=range(len(self._outputs))          
```

Comment:
Two points here:

- The Transaction class is really funny.  A transaction sometimes has no `self._outputs` array defined until you call `self.outputs()` at which time it deserializes the byte vector and then populates all its fields.  Not sure if this is relevant now as you use this function but you may want to make it "safe" and call `self.outputs()` at least once.  
- I recommend doing `for i in range(len(self.outputs())):` here.. It guarantees the transaction gets deserialized .. then you can later safely access `self._outputs` without worry.  Also this eliminates the need for `x` which is not strictly needed as a temporary.